### PR TITLE
Feature/issue 1590, 2031, 1703 Publish and Cancel changes "Доходи та витрати", Publish changes in the Media Settings tab

### PR DESCRIPTION
**Pull request by fhyhtyh**
## Github ticker

* [Github ticket](https://github.com/ita-social-projects/VictoryCenter-Back/issues/1590)
* [Github ticket](https://github.com/ita-social-projects/VictoryCenter-Back/issues/2031)
* [Github ticket](https://github.com/ita-social-projects/VictoryCenter-Back/issues/1703)

## Description

This PR implements the end-to-end Publish and Cancel workflows for the "Доходи та витрати" (Funds and Expenditures) and Media Settings tabs in the Admin Panel. It ensures that admins can safely preview, publish, or revert financial and media data, and synchronizes the public "Звітність" page to dynamically display these published changes (removing legacy hardcoded mock data).

## Summary of change

*   **Publish & Cancel Workflows (`ReportAnalytics.tsx`)**: Introduced confirmation modals for publishing and cancelling draft changes. Successfully syncing these actions with backend endpoints to either persist drafts or restore from backups, followed by reverting the UI to read-only mode and displaying success toasts.
*   **Robust State Reset Pattern (`ReportAnalytics.tsx`)**: Implemented a React key reset strategy (`renderKey`) upon cancellation. This forces a complete unmount and remount of all child tabs, guaranteeing that stale local state and obsolete database IDs are wiped out and pristine data is cleanly refetched.
*   **Media Settings Integration (`MediaSettings.tsx`, `ReportsPanelContent.tsx`)**: Added the publish workflow for Media Settings (titles, changed lives count, and images), complete with validation, confirmation modals, and toast notifications.
*   **Dynamic Public Reports Data (`SummarySection.tsx`)**: Removed hardcoded mock values (e.g., the static 205 "changed lives" value). The public reports page now dynamically pulls real-time text and statistics from the published media settings.
*   **Dynamic CSS Backgrounds (`_effects.mixins.scss`, `StatCard.tsx`)**: Refactored the SCSS `grayscale-bg-image` mixin to accept a `--dynamic-bg-image` CSS custom property. This allows the public site to correctly display newly uploaded images with the grayscale overlay, replacing previously hardcoded local asset paths.
*   **Test Coverage Expansion (`*.test.tsx`)**: Added comprehensive tests for the new publish and cancel modal behaviors, UI edge cases, error handling paths, and API integrations, maintaining high test reliability and coverage.

## How to recreate changes

1. Log into the Admin panel and navigate to the **"Звітність"** (Reports) page.
2. **Publish / Cancel Funds and Expenditures**: Navigate to "Доходи та витрати". Enter edit mode and make a modification to an expense/fund or program record.
3. Click **"Відмінити"** (Cancel) -> verify the confirmation modal appears. Click **"ТАК"** and ensure the changes are discarded, returning the UI to the pristine database data in read-only mode.
4. Make another change and click **"Опублікувати"** (Publish) -> verify the modal, approve it, and ensure a success toast appears.
5. **Publish Media Settings**: Switch to the **"Налаштування медіа"** (Media Settings) tab.
6. Update the titles, "Changed Lives" number, and upload new background images. Click **"Опублікувати"** and confirm.
7. **Verify Public Page**: Open the public **"Звітність"** (Reports) page.
8. Verify that the "Зібрані кошти" (Collected Funds) and "Змінених життів" (Changed Lives) cards now display your newly published numbers, text, and grayscale background images rather than hardcoded defaults.

## CHECK LIST
- [x]  New tests was added or existing was modified
- [ ]  Changes has severe impact on users
- [ ]  Changes has medium impact on users
- [x]  Changes has light impact on users
- [x]  Test covetage was updated
- [x]  PR meets all conventions

## Summary by CodeRabbit

* **New Features**
  * Added publishing and cancellation endpoints for funds and expenditures reports.
  * Added an edit/publish footer with confirmation dialogs and publish availability rules.
  * Added new callbacks to coordinate record counts, data refresh, validation, and unpublished-change state across sections.

* **Bug Fixes**
  * Improved disclaimer/localization synchronization after publish/cancel and save actions.
  * Ensured media input always receives the expected controlled value shape.
  * Updated dynamic background image handling to support CSS custom-property overrides.

* **Tests**
  * Expanded and adjusted test coverage for the new publish/cancel, validation, save, and refresh flows.

* **Chores**
  * Updated styles (added/removed footer styling in relevant report sections).

### CODE REVIEW

File: src/pages/public/reports-page/components/summary-section/SummarySection.tsx [hunk 1]
```diff
@@ -66,11 +120,12 @@ export const SummarySection = () => {
...
                 className={styles.lives}
-                value={SUMMARY_DATA.livesChanged}
-                label={t('summary.lives')}
+                value={data.mediaSettings.changedLives.value || 0}
+                label={data.mediaSettings.changedLives.title || t('summary.lives')}
+                imageUrl={data.mediaSettings.changedLives.imageUrl}
                 color="yellow"
             />
-            {isEn && SUMMARY_DATA.disclaimer && <p className={styles.disclaimer}>{SUMMARY_DATA.disclaimer}</p>}
+            {data.settings.disclaimerTitle && <p className={styles.disclaimer}>{data.settings.disclaimerTitle}</p>}
```
Comment (mehalyna) [inline src/pages/public/reports-page/components/summary-section/SummarySection.tsx]: Previously this was gated on isEn && SUMMARY_DATA.disclaimer — the currency disclaimer only appeared in the English/USD view. It now shows in both locales whenever the backend returns a disclaimerTitle. The test does not render disclaimer when UAH is active still passes only because it was changed to mock disclaimerTitle: null (SummarySection.test.tsx:100), so the test name no longer describes what it verifies. If the intent is "disclaimer shows in any language when present," fine — but the stale test name is misleading and worth renaming.

File: src/pages/admin/reports/components/report-analytics/ReportAnalytics.tsx [hunk 2]
```diff
@@ -49,6 +67,19 @@ export const ReportAnalytics = () => {
...
+    const fetchSettings = useCallback(() => FundsExpendituresApi.getSettings(adminClient), [adminClient]);
+    const { data: settingsData } = useDataFetch({
+        initialData: null,
+        fetchHandler: fetchSettings,
+        autoFetchDependencies: [adminClient],
+    });
+
+    useEffect(() => {
+        if (settingsData) {
+            setHasUnpublishedChanges(settingsData.hasUnpublishedChanges);
```
Comment (mehalyna) [inline src/pages/admin/reports/components/report-analytics/ReportAnalytics.tsx]: It now calls FundsExpendituresApi.getSettings while FundsExpendituresSection.tsx:202 fetches the same settings independently. Two consequences:
- Redundant network request on load.
- hasUnpublishedChanges lives in ReportAnalytics state, seeded once from settingsData, then flipped to true via onDataChange and to false after publish/cancel — but never re-synced with the child. After a successful publish, the parent sets hasUnpublishedChanges=false locally without refetching, so if the backend's real flag diverges (e.g. publish partially succeeds server-side) the UI and server disagree. Low probability, but the split ownership is fragile.

File: src/pages/admin/reports/components/report-analytics/ReportAnalytics.tsx [hunk 3]
```diff
@@ -115,11 +196,26 @@ export const ReportAnalytics = () => {
...
+                        onValidationChange={setIsFundsValid}
+                        onCountsChange={(counts) => {
+                            setIncomeCount(counts.income);
+                            setExpenseCount(counts.expense);
+                        }}
+                        onDataChange={() => setHasUnpublishedChanges(true)}
+                        registerSaveCallback={(saveFn) => {
+                            saveSettingsCallbackRef.current = saveFn;
+                        }}
                     />
```
Comment (mehalyna) [inline src/pages/admin/reports/components/report-analytics/ReportAnalytics.tsx:201]: Both sections are rendered with display: none toggling instead of conditional mount. This is presumably deliberate (so onCountsChange keeps programRecordsCount fed for isPublishEnabled even when the tab isn't active). The tradeoff: the program-expenses data now fetches on initial page load regardless of tab. Worth confirming that's acceptable; it's a real change from lazy-on-tab behavior.

File: src/pages/public/reports-page/components/summary-section/stat-card/StatCard.tsx [hunk 4]
```diff
@@ -29,8 +38,12 @@ export const StatCard = ({ value, label, color = 'blue', currency, formattedValu
     const rootClasses = cn(styles.card, className);
     const valueClasses = cn(styles.value, styles[`text-${color}`]);
 
+    const backgroundStyle = imageUrl
+        ? ({ '--dynamic-bg-image': `url('${imageUrl}')` } as React.CSSProperties)
```
Comment (mehalyna) [inline src/pages/public/reports-page/components/summary-section/stat-card/StatCard.tsx]: Not an XSS vector (custom-property substitution can't escape the background-image declaration, and javascript: URLs don't execute in background-image), but a single quote or paren in imageUrl will produce a malformed url() and silently drop the background. Since this value comes from the backend, encoding the URL (or at least escaping ') would make it robust. Low priority given a trusted source.

File: src/pages/public/reports-page/components/summary-section/expenses-breakdown-chart/use-chart-geometry/useChartGeometry.ts [hunk 5]
```diff
@@ -1,22 +1,23 @@
...
-            const { dx, dy } = LABEL_OFFSETS[index];
+            const { dx, dy } = LABEL_OFFSETS[index] || { dx: 0, dy: 0 };
 
             const bbox = path.getBBox();
             const centerX = bbox.x + bbox.width / 2;
             const centerY = bbox.y + bbox.height / 2;
 
             const length = path.getTotalLength();
-            const point = path.getPointAtLength(length);
+            const targetLength = length * (Math.max(0.01, percents[index] || 0) / 100);
```
Comment (mehalyna) [inline src/pages/public/reports-page/components/summary-section/expenses-breakdown-chart/use-chart-geometry/useChartGeometry.ts:19]: a genuine 0% slice gets a 0.01 floor for label placement. Correct for avoiding a degenerate point; just noting it's intentional. The LABEL_OFFSETS[index] || { dx: 0, dy: 0 } guard is a good defensive addition for when item count exceeds configured offsets.

File: src/pages/admin/reports/components/block-component/ReportsMediaBlock.tsx [hunk 6]
```diff
@@ -206,7 +206,7 @@ export const ReportsMediaBlock = ({
                             variant="partnerBanner"
                             label={COMMON_TEXT_ADMIN.INPUT.ADD_FILE_HERE}
                             subText={COMMON_TEXT_ADMIN.INPUT.getImageSizeSubText(imageWidth, imageHeight)}
-                            value={values.image && 'base64' in values.image ? values.image : null}
+                            value={values.image}
```
Comment (mehalyna) [inline src/pages/admin/reports/components/block-component/ReportsMediaBlock.tsx:193]: ImageInput value fix — dropping the 'base64' in values.image guard so server images (with url) now display is a correct fix; ImageInput's effect already handles the 'url' in value branch. Good.

File: src/pages/admin/reports/components/funds-expenditures-section/FundsExpendituresSection.tsx [hunk 7]
```diff
@@ -652,33 +679,35 @@ export const FundsExpenditureSection = ({
...
             fetchDisclaimerLocalizations(updatedSettings.id);
 
-            addToast(COMMON_TEXT_ADMIN.MESSAGE.SUCCESSFULLY_PUBLISHED, ToastType.Success);
-
             try {
                 await refetchSettings(true);
-            } catch {
-                // Refetch error handled by useDataFetch
-            }
+            } catch {}
```
Comment (mehalyna) [inline src/pages/admin/reports/components/funds-expenditures-section/FundsExpendituresSection.tsx:701]: Empty catch (error) blocks in handlePublishConfirm/handleCancelConfirm use the bound error variable only for a generic toast — fine, but ESLint may flag the unused binding depending on config (the branch already has a lint-fix commit, so presumably OK).

File: src/pages/admin/reports/components/report-analytics/ReportAnalytics.tsx (hunk 3 above)
Comment (fhyhtyh) [reply src/pages/admin/reports/components/report-analytics/ReportAnalytics.tsx:201]: Using display: none prevents the child components from unmounting when switching tabs, which is crucial for preserving local unsaved form state. It also ensures that the "Publish" button logic (which requires at least one program record to be present) can be accurately evaluated without forcing the user to manually visit the "Program Expenses" tab first.

**Review by mehalyna — APPROVED** [review]
